### ISSUE TITLE
`mutate_all()` generates wrong sql statement

### ISSUE THREAD

**Issue by wanting0wang** [sparklyr/sparklyr#3085]
While running some code on sparklyr 1.5.1, I got wrong result for `mutate_all(is.na)`. It is not working and always returns FALSE even when there is a missing value.

Here is a reprex:
```
tbl_data <- copy_to(sc,data.frame(V1=c(1,2,NA)),'test',overwrite=T)

x <- tbl_data %>%
  mutate(V1 = is.na(V1))
x

y <- tbl_data %>%
  mutate_all(is.na)
y
```

Result as follows:
![image](https://user-images.githubusercontent.com/25274600/120086808-1ee47d80-c0b0-11eb-9834-a5fd28bd9197.png)

Further looking into the sql statement generated, apparently `mutate_all(is.na)` generates a wrong statement:
```
> dbplyr::sql_render(x)
<SQL> SELECT ((`V1`) IS NULL) AS `V1`
FROM `test`
> dbplyr::sql_render(y)
<SQL> SELECT ((`V1`) IS NULL) AS `V1`
FROM (SELECT ((`V1`) IS NULL) AS `V1`
FROM `test`) `q01`
```
The `mutate_all(is.na)` tries to do `is.na` twice in the sql command, leading the result to have 0 missing values.

Session info:
```
R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    
system code page: 936

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_1.0.3    sparklyr_1.5.1

loaded via a namespace (and not attached):
 [1] pillar_1.4.7      compiler_4.0.3    dbplyr_2.0.0      r2d3_0.2.5       
 [5] base64enc_0.1-3   profvis_0.3.7     tools_4.0.3       digest_0.6.27    
 [9] uuid_0.1-4        jsonlite_1.7.2    lifecycle_1.0.0   tibble_3.0.6     
[13] pkgconfig_2.0.3   rlang_0.4.10      DBI_1.1.1         cli_2.5.0        
[17] rstudioapi_0.13   yaml_2.2.1        parallel_4.0.3    stringr_1.4.0    
[21] withr_2.4.1       httr_1.4.2        generics_0.1.0    vctrs_0.3.6      
[25] htmlwidgets_1.5.3 askpass_1.1       rappdirs_0.3.3    rprojroot_2.0.2  
[29] tidyselect_1.1.0  glue_1.4.2        forge_0.2.0       R6_2.5.0         
[33] fansi_0.4.2       purrr_0.3.4       tidyr_1.1.2       blob_1.2.1       
[37] magrittr_2.0.1    ellipsis_0.3.1    htmltools_0.5.1.1 assertthat_0.2.1 
[41] config_0.3.1      utf8_1.1.4        stringi_1.5.3     openssl_1.4.3    
[45] crayon_1.4.0     
```

**Comment by wanting0wang** [comment]
I see a very similar issue is mentioned in https://github.com/tidyverse/dbplyr/issues/605 and the problem might be solved in https://github.com/sparklyr/sparklyr/pull/2962 at least for `transmute()`.

For the work that has to be done on sparklyr 1.5.1, how can we apply a hot fix? Thank you.

**Comment by wanting0wang** [comment]
Based on 1.5.1 (https://github.com/sparklyr/sparklyr/commit/8a5f00f830cfdfd1718a4673abed30a644eefcbe), I cherry picked the two commits (https://github.com/sparklyr/sparklyr/commit/a84e00ddf08e400b1f1fdea5097580687e182389 and https://github.com/sparklyr/sparklyr/commit/d009cc2fc9a50faa51053c26727cd0d3a49c8066) that are said to fix the duplicated sql string issue.

I'm able to get the right result for `mutate_all(is.na)` (put version 1.5.2 for this):

![image](https://user-images.githubusercontent.com/25274600/120093874-0f365a80-c0eb-11eb-98a1-7e188bad42d2.png)

@yitao-li do you see any possible issue with this?

**Comment by yitao-li** [comment]
@wanting0wang The hotfix looks alright to me. Although I do recommend switching to `sparklyr` 1.6.2 instead rather than hotfixing an older version of `sparklyr`.

**Comment by wanting0wang** [comment]
@yitao-li Thanks for the confirmation! 
We are looking at 1.5.1 again after trying out 1.6.2 because of https://github.com/sparklyr/sparklyr/issues/3009 - some data loading process that used to take 2-3min has to take 15-25min on 1.6.2; I'm thinking about providing a patched 1.5.1 (cherry pick the 2 commits mentioned above) and a patched 1.6.2 (cherry pick the 4 commits mentioned in https://github.com/sparklyr/sparklyr/issues/3009) to the users.
